src/intrinsics.cpp

```diff
@@ -70,85 +106,80 @@ extern "C" JL_DLLEXPORT int8_t jl_is_memdebug() {
     where the box is needed.
 */
 
-static Type *FTnbits(size_t nb)
+// convert an llvm type to same-size float type
+static Type *FLOATT(Type *t)
 {
+    if (t->isFloatingPointTy())
+        return t;
+    unsigned nb = (t->isPointerTy() ? t->getPrimitiveSizeInBits() : sizeof(void*));
+    if (nb == 64)
+        return T_float64;
+    if (nb == 32)
+        return T_float32;
 #ifndef DISABLE_FLOAT16
     if (nb == 16)
         return T_float16;
-    else
 #endif
-    if (nb == 32)
-        return T_float32;
-    else if (nb == 64)
-        return T_float64;
-    else if (nb == 128)
+    if (nb == 128)
         return T_float128;
-    else
-        jl_error("Unsupported Float Size");
-}
-// convert int type to same-size float type
-static Type *FT(Type *t)
-{
-    if (t->isFloatingPointTy())
-        return t;
-    return FTnbits(t->getPrimitiveSizeInBits());
+    jl_error("Unsupported Float Size");
 }
 
-// reinterpret-cast to float
-static Value *FP(Value *v)
-{
-    if (v->getType()->isFloatingPointTy())
-        return v;
-    return emit_bitcast(v, FT(v->getType()));
-}
-
-// convert float type to same-size int type
-static Type *JL_INTT(Type *t)
+// convert an llvm type to same-size int type
+static Type *INTT(Type *t)
 {
     if (t->isIntegerTy())
         return t;
     if (t->isPointerTy())
         return T_size;
-    if (t == T_float32) return T_int32;
-    if (t == T_float64) return T_int64;
-    assert(t == T_void);
-    return T_void;
+    if (t == T_float64)
+        return T_int64;
+    if (t == T_float32)
+        return T_int32;
+    if (t == T_float16)
+        return T_int16;
+    //if (t == T_float128)
+    //    return T_int128;
```

Comment:
leave a comment or a cross-reference about why this is commented out (same below in `JL_JL_UINTT`)